Project: Building with Active Record

george-swift opened the pull request:
## Building Micro-Reddit
The project required data modeling and putting our active record knowledge to practice by building a lightweight Reddit clone. Three data models were generated for User, Post, and Comment using the `rails generate model` command. 
The posts and comments are associated using the `references` with the User using the `has_many` / `belongs_to` Active Record method. Validations were also added for comments, posts, and users.

MiguelArgentina committed:
Initial Commit. Add linters

MiguelArgentina committed:
Update README.md

MiguelArgentina committed:
Add nio4r gem

MiguelArgentina committed:
Create User model

MiguelArgentina committed:
Migrate db

MiguelArgentina committed:
Create a test for User model

MiguelArgentina committed:
Create Post model

MiguelArgentina committed:
Add association to Post model

george-swift committed:
Add association to post model

george-swift committed:
Add test for a user's post

george-swift committed:
Add test to validate post references user

george-swift committed:
Add tests to check post title validation

george-swift committed:
Add tests to validate the body of each post

MiguelArgentina committed:
Add association with comments

MiguelArgentina committed:
Create Comments model

MiguelArgentina committed:
Update README.md

MiguelArgentina committed:
Correct linter errors

MiguelArgentina committed:
Update README.md

MiguelArgentina committed:
Changing branch due to commits history problems

Georjane reviewed (changes requested):
## Status: Changes Required 🛑 🔧 
### Hi there Team ✋🏽 
Very nice work you have done with the micro Reddit project. All your associations are working just fine. We however have a few changes to make to improve on this project:

- [x] In your models, you don't need validations for the user_id or post_id. Once you use the belongs_to user, it already takes care of the validation for user_id.
for example, instead of this, 
```
class Comment < ApplicationRecord
  belongs_to :user
  belongs_to :post

  validates :user_id, presence: true
  validates :post_id, presence: true
  validates :body, presence: true, length: { minimum: 2 }
end
```
you could have just this,
```
class Comment < ApplicationRecord
  belongs_to :user
  belongs_to :post

  validates :body, presence: true, length: { minimum: 2 }
end
```

- [x] Also you can improve your Readme file to look more professional by adding instructions for setting up this project such as using `git clone ... ` and `bundle install`
- [x] In your Read me file, it is nice you added a link to the actual project. But you should also include a few commands to test your associations such as:
```
> User.all
> u1 = User.new(name: 'users', nickname: 'usersnick')
> User.first.posts
```
just to name a few.

Fix the above issues and request another review.

_Feel free to reach out to me on Slack if you have any questions, Witah Georjane_ 🤝 
Good luck on your project and Happy Coding 💻 

**Witah Georjane
Code Reviewer at Microverse Inc
[portfolio](https://witahgeorjane.tech/) | [linkedIn](https://www.linkedin.com/in/witah-georjane/)**

george-swift committed:
Remove redundant validations

In the models, validations for the user_id or post_id are not needed. After associating with belongs_to user, it already takes care of the validation for user_id. As seen in db/migrate file

george-swift committed:
Add screenshot to assets

george-swift committed:
Modify README

george-swift committed:
Modify README

george-swift committed:
Add screenshot

oloomoses reviewed (approved):
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_

oloomoses reviewed (approved):
# Congratulations, Project Approved. :muscle: :tada: :clap: 
**Great job working on the suggested changes from the previous review.
You have passed all the requirements for this project :muscle: 
Feel free to merge and close this pull request.**

**I wish you the best in your next project and see you at the top :rocket:** 
#### Happy Coding
_Slack: Oloo Moses_
